nautobot_secrets_providers/tests/test_providers.py

```diff
@@ -199,6 +253,198 @@ def test_retrieve_invalid_key(self, requests_mocker):
+
+    @requests_mock.Mocker()
+    @patch("builtins.open", new_callable=mock_open, read_data="data")
+    def test_get_client_k8s(self, requests_mocker, mock_file):
+        """Test Kubernetes Authentication."""
+        vault_url = "http://localhost:8200"
+        k8s_token_path = "/some/file/path"  # nosec B105
+        new_plugins_config = {
+            "nautobot_secrets_providers": {
+                "hashicorp_vault": {
+                    "url": vault_url,
+                    "auth_method": "kubernetes",
+                    "k8s_token_path": k8s_token_path,
+                },
+            },
+        }
+
+        # Test without specifying a role_name
+        with self.settings(PLUGINS_CONFIG=new_plugins_config):
+            with self.assertRaises(exceptions.SecretProviderError) as err:
+                self.provider.get_client(self.secret)
+        self.assertEqual(
+            str(err.exception),
+            'SecretProviderError: Secret "hello-hashicorp" (provider "HashiCorpVaultSecretsProvider"): HashiCorp Vault configuration is missing a role name for kubernetes authentication!',
+        )
+
+        # Test with various response codes (https://www.vaultproject.io/api-docs#http-status-codes)
+        new_plugins_config["nautobot_secrets_providers"]["hashicorp_vault"]["role_name"] = "some_role"
+        with self.settings(PLUGINS_CONFIG=new_plugins_config):
+            # Test Valid Response
+            requests_mocker.register_uri(
+                method="POST",
+                url=f"{vault_url}/v1/auth/kubernetes/login",
+                status_code=200,
+                json=self.mock_kubernetes_auth_response,
+            )
+            hvac_client = self.provider.get_client(self.secret)
+            self.assertIsInstance(hvac_client, HVACClient)
+
+            # Test Invalid Credentials
+            requests_mocker.register_uri(method="POST", url=f"{vault_url}/v1/auth/kubernetes/login", status_code=403)
+            with self.assertRaises(exceptions.SecretProviderError) as err:
+                self.provider.get_client(self.secret)
+            self.assertEqual(
+                str(err.exception),
+                'SecretProviderError: Secret "hello-hashicorp" (provider "HashiCorpVaultSecretsProvider"): HashiCorp Vault Access Denied (auth_method: kubernetes). Error: , on post http://localhost:8200/v1/auth/kubernetes/login',
+            )
+
+            # Test Invalid Request
+            requests_mocker.register_uri(method="POST", url=f"{vault_url}/v1/auth/kubernetes/login", status_code=400)
+            with self.assertRaises(exceptions.SecretProviderError) as err:
+                self.provider.get_client(self.secret)
+            self.assertEqual(
+                str(err.exception),
+                'SecretProviderError: Secret "hello-hashicorp" (provider "HashiCorpVaultSecretsProvider"): HashiCorp Vault Login failed (auth_method: kubernetes). Error: , on post http://localhost:8200/v1/auth/kubernetes/login',
```

Comment:
Is it expected that in actual practice we will have a blank `Error: ,` text in the exception? Or is the request mock just not providing an accurate representation of the error we'd get in practice?

Reply:
The mock isn't providing a correct error response